internal/database/migration/stitch/rewriter.go

```diff
@@ -0,0 +1,316 @@
+			return strings.ReplaceAll(oldQuery, " CONCURRENTLY", "")
+		})
+	}
+}
+
+// reorderMigrations reproduces an explicit (historic) reodering of several migration files. For versions where
+// these files exist and haven't yet been renamed, we do the renaming at this time to make it match later versions.
+//
+// See https://github.com/sourcegraph/sourcegraph/pull/29395.
+func reorderMigrations(schemaName string, minorVersion int, _ []int, contents map[string]string) {
+	if schemaName != "frontend" || minorVersion >= 36 {
+		// Rename occurred at v3.36
+		return
+	}
+
+	for oldID, newID := range map[int]int{
+		1528395945: 1528395961,
+		1528395946: 1528395962,
+		1528395947: 1528395963,
+		1528395948: 1528395964,
+	} {
+		if _, ok := contents[migrationFilename(oldID, "metadata.yaml")]; !ok {
+			// File doesn't exist at this verson (nothing to rewrite)
+			continue
+		}
+
+		// Move new contents and replace previous contents
+		noopContents := "-- NO-OP to fix out of sequence migrations"
+		contents[migrationFilename(newID, "up.sql")] = contents[migrationFilename(oldID, "up.sql")]
+		contents[migrationFilename(newID, "down.sql")] = contents[migrationFilename(oldID, "down.sql")]
+		contents[migrationFilename(oldID, "up.sql")] = noopContents
+		contents[migrationFilename(oldID, "down.sql")] = noopContents
+
+		// Write new metadata
+		oldMetadata := contents[migrationFilename(oldID, "metadata.yaml")]
+		contents[migrationFilename(newID, "metadata.yaml")] = replaceParents(oldMetadata, newID-1)
+	}
+}
+
+func idsFromRawMigrations(rawMigrations []rawMigration) ([]int, error) {
+	ids := make([]int, 0, len(rawMigrations))
+	for _, rawMigration := range rawMigrations {
+		id, err := strconv.Atoi(rawMigration.id)
+		if err != nil {
+			return nil, err
+		}
+
+		ids = append(ids, id)
+	}
+
+	sort.Ints(ids)
+	return ids, nil
+}
+
+func migrationFilename(id int, filename string) string {
+	return filepath.Join(strconv.Itoa(id), filename)
+}
+
+// mapContents transforms and replaces the contents of the given filename, if it is already present in the map.
+// An absent entry results in a no-op.
+func mapContents(contents map[string]string, filename string, f func(v string) string) {
+	if v, ok := contents[filename]; ok {
+		contents[filename] = f(v)
+	}
+}
+
+var yamlParentsPattern = lazyregexp.New(`parents: \[[\d,]+\]`)
+
+// removeParents removes the `parents: ` line from the given YAML file contents.
+func removeParents(contents string) string {
+	return yamlParentsPattern.ReplaceAllString(contents, "")
+}
+
+// replacesParents removes the `parents: ` line from the given YAML file contents and inserts a new line with the
+// given parent identifiers.
+func replaceParents(contents string, parents ...int) string {
+	strParents := make([]string, 0, len(parents))
+	for _, id := range parents {
+		strParents = append(strParents, strconv.Itoa(id))
+	}
+
+	return removeParents(contents) + fmt.Sprintf("\nparents: [%s]", strings.Join(strParents, ", "))
```

Comment:
This matches the `yamlParentsPattern`:

```suggestion
	return removeParents(contents) + fmt.Sprintf("parents: [%s]", strings.Join(strParents, ","))
```

Reply:
contents does not necessarily end in a newline